src/analysis/inner_product_space/gram_schmidt_ortho.lean

```diff
@@ -0,0 +1,342 @@
+            ((⟪gram_schmidt 𝕜 f c₁, f (c + 1)⟫
+              / ∥gram_schmidt 𝕜 f c₁∥ ^ 2) : 𝕜) hc₃ } } },
+    have h₁ : gram_schmidt 𝕜 f '' Iic c.succ ⊆ span 𝕜 (f '' Iic c.succ) := h₀,
+    rwa ← span_le at h₁ },
+  { have h₀ : ∀ x ∈ f '' Iic c.succ,
+      x ∈ span 𝕜 (gram_schmidt 𝕜 f '' Iic c.succ),
+    { intros x hx,
+      simp only [mem_image, mem_Iic] at hx,
+      rcases hx with ⟨y, hy₁, hy₂⟩,
+      rw nat.succ_eq_add_one at hy₁ ⊢,
+      by_cases y ≤ c,
+      { have hx : x ∈ f '' Iic c := ⟨y, h, hy₂⟩,
+        have hx₁ : x ∈ span 𝕜 (f '' Iic c),
+        { have h₁ : f '' Iic c ⊆ span 𝕜 (f '' Iic c) := subset_span,
+          exact h₁ hx },
+        rw ← hc at hx₁,
+        have h₂ : span 𝕜 (gram_schmidt 𝕜 f '' Iic c)
+          ≤ span 𝕜 (gram_schmidt 𝕜 f '' Iic (c + 1)),
+        { have h₃ : gram_schmidt 𝕜 f '' Iic c ⊆ gram_schmidt 𝕜 f '' Iic (c + 1),
+          { have h₄ : Iic c ⊆ Iic (c + 1),
+            { simp only [Iic, set_of_subset_set_of],
+              intros a ha,
+              linarith },
+            exact image_subset (gram_schmidt 𝕜 f) h₄ },
+          exact span_mono h₃ },
+        exact h₂ hx₁ },
+      { have hy₃ : y = c + 1 := by linarith,
+        rw hy₃ at hy₂,
+        have hx : x = gram_schmidt 𝕜 f (c + 1) + ∑ i in finset.range (c + 1),
+          orthogonal_projection (𝕜 ∙ gram_schmidt 𝕜 f i) x :=
+            by simp only [gram_schmidt_def, hy₂, sub_add_cancel],
+        simp only [orthogonal_projection_singleton] at hx,
+        rw hx,
+        apply add_mem _ _ _,
+        { have h₁ : gram_schmidt 𝕜 f (c + 1) ∈ gram_schmidt 𝕜 f '' Iic c.succ,
+          { simp only [mem_image, mem_Iic],
+            refine ⟨c + 1, by simp only [nat.succ_eq_add_one], by refl⟩ },
+          have h₂ : gram_schmidt 𝕜 f '' Iic c.succ
+            ⊆ span 𝕜 (gram_schmidt 𝕜 f '' Iic c.succ) := subset_span,
+          exact h₂ h₁ },
+        { apply sum_mem _ _,
+          intros c₁ hc,
+          have hc₁ : gram_schmidt 𝕜 f c₁ ∈
+            span 𝕜 (gram_schmidt 𝕜 f '' Iic c.succ),
+          { have hc₂ : gram_schmidt 𝕜 f c₁ ∈ gram_schmidt 𝕜 f '' Iic c.succ,
+            { simp only [mem_image, mem_Iic, nat.succ_eq_add_one],
+              refine ⟨c₁, _, by refl⟩,
+              rw finset.mem_range at hc,
+              linarith },
+            have hc₃ : gram_schmidt 𝕜 f '' Iic c.succ ⊆
+              span 𝕜 (gram_schmidt 𝕜 f '' Iic c.succ) := subset_span,
+            exact hc₃ hc₂ },
+          exact smul_mem (span 𝕜 (gram_schmidt 𝕜 f '' Iic (c + 1)))
+            ((⟪gram_schmidt 𝕜 f c₁, x⟫ / ∥gram_schmidt 𝕜 f c₁∥ ^ 2) : 𝕜)
+              hc₁ } } },
+    have h₁ : f '' Iic c.succ ⊆ span 𝕜 (gram_schmidt 𝕜 f '' Iic c.succ) := h₀,
+    rwa ← span_le at h₁ }
+end
+
+/-- If the input of first n vectors of gram_schmidt are linearly independent
+,then output of first n vectors are non-zero -/
+lemma gram_schmidt_ne_zero (f : ℕ → E) (n : ℕ)
+  (h₀ : linear_independent 𝕜 (f ∘ (coe : fin n.succ → ℕ))) :
```

Comment:
This is requiring linear independence of the first `n+1` vectors, which is not what the docstring says.

Reply:
oh yes, I am sorry for that. I have changed doc here